c_glib/arrow-glib/compute.cpp

```diff
@@ -37,6 +37,9 @@
 #include <arrow/acero/exec_plan.h>
 #include <arrow/acero/options.h>
 
+// Initialize the compute library and register compute kernels.
+auto compute_init_status_ = arrow::compute::Initialize();
```

Comment:
This makes it a public symbol, should it be private instead?
```suggestion
static auto compute_init_status_ = arrow::compute::Initialize();
```

Reply:
@kou I don't know much about the CGLib implementation. What would be the best way to check for the status here? Should I add a function like:
```
gboolean
garrow_compute_is_initialized(void)
{
  return compute_init_status_.ok();
}
```
but how would we check this? Should I validate the initialization on the different methods that require compute?

Reply:
If we need to check the status, we must add an initialization function (something like `gboolean garrow_compute_initialize(GError **error)`) and users call it explicitly.

BTW, is there possibility that `arrow::compute::Initialize()` is failed? If `arrow::compute::Initialize()` is never failed, we don't need to check the status and we may use different return type than `arrow::Status`.

Reply:
we were always returning `Status::OK()` and the current function registry does not return or checks for errors registering so having `arrow::compute::Initialize()` not return `arrow::Status` makes sense to me.
@pitrou @kou what do you think about this, I've pushed a commit with the new behavior:
https://github.com/apache/arrow/pull/46261/commits/13a20f02796e6bf38e68fb84f314d8faa61db350

Reply:
Returning `Status` is more future-proof if we add more steps that may actually fail.

(also, registering the functions _can_ fail, it's just that we currently don't propagate the errors)

Reply:
Thanks @pitrou , I've reverted the previous commit.
@kou I've pushed the following c_glib change:
```c
// Initialize the compute library and register compute kernels.
static auto compute_init_status_ = arrow::compute::Initialize();

gboolean
garrow_compute_is_initialized(GError **error)
{
  if (!compute_init_status_.ok()) {
    g_set_error(error,
                GARROW_ERROR,
                GARROW_ERROR_UNKNOWN,
                "Arrow compute module is not properly initialized: %s",
                compute_init_status_.ToString().c_str());
  }
  return compute_init_status_.ok();
}
```
I can update this to a function to make the explicit call to Initialize but I am not sure how can I make our Ruby implementation do the call implicitly? do we have some initialization already? 

Or could be enough providing a function to users so they can check the status of the initialization? as the last commit

Reply:
Can I put a commit to this branch for it?

Reply:
Oh, sorry. I missed your previous comment... I'll do it now.